src/modals/WalletConnect/WalletConnectSessionProposalModal.tsx

```diff
@@ -0,0 +1,200 @@
+    rejectProposal()
+    onClose()
+
+    posthog.capture('Rejected WalletConnect connection by closing modal')
+  }
+
+  return (
+    <CenteredModal
+      title={
+        metadata.description ? t('Connect to {{ dAppUrl }}', { dAppUrl: cleanUrl(metadata.url) }) : t('Connect to dApp')
+      }
+      subtitle={metadata.description || metadata.url}
+      onClose={rejectAndCloseModal}
+      Icon={() =>
+        metadata?.icons &&
+        metadata.icons.length > 0 &&
+        metadata.icons[0] && <AssetLogo assetImageUrl={metadata.icons[0]} assetId={metadata.url} size={50} />
+      }
+    >
+      {showNetworkWarning ? (
+        <>
+          <Section>
+            <InfoBox label="Switch network" Icon={AlertTriangle}>
+              You are currently connected to <Highlight>{currentNetworkName}</Highlight>, but the dApp requires a
+              connection to <Highlight>{requiredChainInfo?.networkId}</Highlight>.
+            </InfoBox>
+          </Section>
+          <ModalFooterButtons>
+            <ModalFooterButton role="secondary" onClick={rejectProposal}>
+              {t('Decline')}
+            </ModalFooterButton>
+            <ModalFooterButton onClick={handleSwitchNetworkPress}>{t('Switch network')}</ModalFooterButton>
+          </ModalFooterButtons>
+        </>
+      ) : !signerAddress ? (
+        <>
+          <Section>
+            <InfoBox label="New address needed" Icon={PlusSquare}>
+              The dApp asks for an address in group <Highlight> {requiredChainInfo?.addressGroup}</Highlight>. Click
+              below to generate one!
+            </InfoBox>
+          </Section>
+          <ModalFooterButtons>
+            <ModalFooterButton role="secondary" onClick={rejectProposal}>
+              {t('Decline')}
+            </ModalFooterButton>
+            <ModalFooterButton onClick={generateAddressInGroup}>{t('Generate new address')}</ModalFooterButton>
+          </ModalFooterButtons>
+        </>
+      ) : (
+        <>
+          <Paragraph style={{ marginBottom: 0 }}>
+            {group === undefined ? (
+              <Trans
+                t={t}
+                i18nKey="walletConnectProposalMessage"
+                values={{
+                  dAppUrl: cleanUrl(metadata.url)
+                }}
+                components={{
+                  1: <Highlight />
+                }}
+              >
+                {'Connect to <1>{{ dAppUrl }}</1> with one of your addresses:'}
+              </Trans>
+            ) : (
+              <Trans
+                t={t}
+                i18nKey="walletConnectProposalMessageWithGroup"
+                values={{
+                  dAppUrl: cleanUrl(metadata.url),
+                  group
+                }}
+                components={{
+                  1: <Highlight />,
+                  3: <Highlight />
+                }}
+              >
+                {'Connect to <1>{{ dAppUrl }}</1> with one of your addresses in group <3>{{ group }}</3>:'}
```

Comment:
Question: can't the same component (1) be reused in 2 spots?

Reply:
Hm, good question, haven't tried it.